tests/test_common.py

```diff
@@ -0,0 +1,482 @@
+"""
+<Program Name>
+  test_common.py
+
+<Purpose>
+  Unit and integration testing for uptane/common.py
+
+<Copyright>
+  See LICENSE for licensing information.
+"""
+from __future__ import unicode_literals
+
+import uptane # Import before TUF modules; may change tuf.conf values.
+
+import unittest
+import os.path
+import shutil # for rmtree
+import copy
+import json
+
+import tuf
+import tuf.formats
+import tuf.conf
+
+import uptane.common as common
+import uptane.encoding.asn1_codec as asn1_codec
+
+import demo # for generate_key, import_public_key, import_private_key
+
+# The public and private keys to use during testing, including the Director
+# repository keys (public and private) as well as client keys (public).
+keys_pri = {}
+keys_pub = {}
+
+SAMPLES_DIR = os.path.join(uptane.WORKING_DIR, 'samples')
+TEST_DATA_DIR = os.path.join(uptane.WORKING_DIR, 'tests', 'test_data')
+TEST_DIRECTOR_METADATA_DIR = os.path.join(TEST_DATA_DIR, 'director_metadata')
+TEST_IMAGE_REPO_METADATA_DIR = os.path.join(
+    TEST_DATA_DIR, 'image_repo_metadata')
+TEST_DIRECTOR_ROOT_FNAME = os.path.join(
+    TEST_DIRECTOR_METADATA_DIR, 'root.' + tuf.conf.METADATA_FORMAT)
+TEST_IMAGE_REPO_ROOT_FNAME = os.path.join(
+    TEST_IMAGE_REPO_METADATA_DIR, 'root.' + tuf.conf.METADATA_FORMAT)
+TEST_PINNING_FNAME = os.path.join(TEST_DATA_DIR, 'pinned.json')
+TEMP_CLIENT_DIR = os.path.join(TEST_DATA_DIR, 'temp_test_common')
```

Comment:
nitpick:
I guess this can be renamed `TEST_TEMP_CLIENT_DIR` to be consistent with the others.  However, although these variable names are descriptive, they might be bordering on excessively long.

One issue I have with the TUF codebase is its use of really long variable names, and wish shorter names had been used in many places.

I guess one could argue that prepending TEST_ to the variables names above is redundant, and perhaps eliminating it would improve readability.

```
# These are temporary test directories used by the tests...
DATA_DIR =
DIRECTOR_METADATA_DIR
...
```

The above is an alternative you may consider, but it isn't a big deal and I would be fine with either approach.

Reply:
I'll take a look and go back and revise these across modules (since they're consistent across test modules).